jetty-io/src/main/java/org/eclipse/jetty/io/MappedByteBufferPool.java

```diff
@@ -99,13 +99,13 @@ public MappedByteBufferPool(int factor, int maxQueueLength, Function<Integer, Bu
      */
     public MappedByteBufferPool(int factor, int maxQueueLength, Function<Integer, Bucket> newBucket, long maxHeapMemory, long maxDirectMemory)
     {
-        super(factor, maxQueueLength, maxHeapMemory, maxDirectMemory);
+        super(factor, 0, maxQueueLength, maxHeapMemory, maxDirectMemory, -1, -1);
```

Comment:
The default heap and direct memory should be 0 to use the heuristic, not -1 for unlimited.

Reply:
The MappedByteBufferPool is primarily used by the client which benefits from a RBBP too, I don't see why we would disable the retainability by default.

That -1 vs 0 change you made breaks the pre-existing contract, perhaps you should revert to 0 for heuristic, -1 for unlimited and add -2 for non-retaining retainable pool?